scripts/validate_my_definition.py

```diff
@@ -1,49 +1,50 @@
 #!/usr/bin/env python3
 
-import jsonschema
-import yaml
 from yaml import YAMLError
-from jsonschema.exceptions import ValidationError
-import requests
+from jsonschema.exceptions import ValidationError, RefResolutionError
 from argparse import ArgumentParser
 import glob
-from almdrlib.client import _YamlOrderedLoader
-OPENAPI_SCHEMA_URL = 'https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/schemas/v3.0/schema.json'
+from alsdkdefs import AlertLogicOpenApiValidationException
+import alsdkdefs
+import os
 
 
 def validate_definition(definition_file):
     print(f"Validating {definition_file}")
-    with open(definition_file, "r") as f:
-        spec = f.read()
-    if spec:
-        try:
-            obj = yaml.load(spec, Loader=_YamlOrderedLoader)
-            jsonschema.validate(obj, schema)
-        except YAMLError as e:
-            print(f"Validation has failed - failed to load YAML {e}")
-            exit(1)
-        except ValidationError as e:
-            print(f"Validation has failed - schema validation has failed {e}")
-            exit(1)
-        except TypeError:
-            print(f"Validation has failed - json schema trips over integer keys, please "
-                  f"validate your response codes are not integers, check also other keys are not integers"
-                  f"if any of it are, quote it, '200', '400' etc")
-            exit(1)
-        print("Validation passed")
-    else:
-        print("Input is empty")
+    try:
+        spec = alsdkdefs.load_spec(definition_file)
+        alsdkdefs.validate(spec, definition_file)
+    except YAMLError as e:
+        print(f"Validation has failed - failed to load YAML {e}")
+        exit(1)
+    except ValidationError as e:
+        print(f"Validation has failed - schema validation has failed {e}")
```

Comment:
no logging utils?

Reply:
That's just a console script, i could include logging capabilities, but i don't see any need for it in the current case, since i don't need levels, formatters with time and stuff, just report progress and return exit code and error description.